discovery-provider/integration_tests/tasks/test_index_trending_underground_notification.py

```diff
@@ -0,0 +1,182 @@
+        index_underground_trending_mock(db, list(range(1, 21)))
+        base_time_int = int(round(BASE_TIME.timestamp()))
+        index_trending_underground_notifications(db, base_time_int)
+
+        with db.scoped_session() as session:
+            notifications = (
+                session.query(Notification).order_by(asc(Notification.specifier)).all()
+            )
+            assert len(notifications) == 5
+
+            notification_0 = notifications[0]
+            assert notification_0.specifier == "1"
+            assert (
+                notification_0.group_id
+                == "trending_underground:time_range:week:genre:all:rank:1:track_id:1:timestamp:1672531200"
+            )
+            assert notification_0.type == "trending_underground"
+            assert notification_0.data == {
+                "rank": 1,
+                "genre": "all",
+                "track_id": 1,
+                "time_range": "week",
+            }
+            assert notification_0.user_ids == [2]
+
+            notification_1 = notifications[1]
+            assert notification_1.specifier == "2"
+            assert (
+                notification_1.group_id
+                == "trending_underground:time_range:week:genre:all:rank:2:track_id:2:timestamp:1672531200"
+            )
+            assert notification_1.type == "trending_underground"
+            assert notification_1.data == {
+                "rank": 2,
+                "genre": "all",
+                "track_id": 2,
+                "time_range": "week",
+            }
+            assert notification_1.user_ids == [1]
+
+        # Test that on second iteration of trending, only new notifications appear
+        # Prev: 1, 2, 3, 4, 5
+        # New: 2, 4, 5, 1, 7
+        # new notifications for track_id: 2, 4, 5, 7
+        # no new notification for track_id: 1
+
+        index_underground_trending_mock(db, [2, 4, 5, 1, 7, 8, 9, 10])
+        updated_time = BASE_TIME + timedelta(hours=1)
+        updated_time_int = int(round(updated_time.timestamp()))
+        index_trending_underground_notifications(db, updated_time_int)
+
+        with db.scoped_session() as session:
+            all_notifications = session.query(Notification).all()
+            assert len(all_notifications) == 6
+
+            updated_notifications = (
+                session.query(Notification)
+                .filter(Notification.timestamp == updated_time)
+                .order_by(asc(Notification.specifier))
+                .all()
+            )
+            assert len(updated_notifications) == 1
+
+            assert updated_notifications[0].data == {
+                "rank": 5,
+                "genre": "all",
+                "track_id": 7,
+                "time_range": "week",
+            }
+
+        updated_time = BASE_TIME + timedelta(hours=24)
+        updated_time_int = int(round(updated_time.timestamp()))
+        index_trending_underground_notifications(db, updated_time_int)
```

Comment:
not sure if this last test case is related to running trending twice beforehand, if not this could be a separate test perhaps (if it is related feel free to ignore)

Reply:
It seems like it actually might be, basically saying that after 24 hours if you end up trending again, then we send you a notif